src/plugins/keras_support/resize_operation.cpp

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) 2019 Shahrzad Shirzad
+// Copyright (c) 2018-2019 Hartmut Kaiser
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <phylanx/config.hpp>
+#include <phylanx/ir/node_data.hpp>
+#include <phylanx/plugins/keras_support/resize_operation.hpp>
+
+#include <hpx/include/lcos.hpp>
+#include <hpx/include/naming.hpp>
+#include <hpx/include/util.hpp>
+#include <hpx/throw_exception.hpp>
+
+#include <algorithm>
+#include <cmath>
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <blaze/Math.h>
+#if defined(PHYLANX_HAVE_BLAZE_TENSOR)
+#include <blaze_tensor/Math.h>
+#endif
+
+///////////////////////////////////////////////////////////////////////////////
+namespace phylanx { namespace execution_tree { namespace primitives
+{
+    ///////////////////////////////////////////////////////////////////////////
+    match_pattern_type const resize_operation::match_data = {
+        hpx::util::make_tuple("resize_images",
+            std::vector<std::string>{"resize_images(_1, _2, _3, _4)"},
+            &create_resize_operation, &create_primitive<resize_operation>,
+            R"(a
+            Args:
+
+                a (array_like) : input array
+                height_factor (integer) : enlargement factor for height
+                width_factor (integer) : enlargement factor for width
+                interpolation (string) : interpolation type, "bilinear" or "nearest"
+
+            Returns:
+
+            Returns a resized version of the array with height_factor and width_factor
+            using the specified interpolation .)")};
+
+    ///////////////////////////////////////////////////////////////////////////
+    resize_operation::resize_operation(primitive_arguments_type&& operands,
+        std::string const& name, std::string const& codename)
+      : primitive_component_base(std::move(operands), name, codename)
+    {
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    template <typename T>
+    primitive_argument_type resize_operation::nearest(ir::node_data<T>&& arg,
+        std::int64_t height_factor, std::int64_t width_factor,
+        std::string interpolation) const
+    {
+        auto m = arg.matrix();
+        blaze::DynamicMatrix<T> result(
+            m.rows() * height_factor, m.columns() * width_factor);
+
+        for (std::size_t i = 0;
+             i < m.rows() * height_factor * m.columns() * width_factor;
```

Comment:
The upper limit for this loop could be calculated once, or simply use `result.size()`